File: Source/iap/Source/Library/iap/src/main/java/com/philips/cdp/di/iap/model/SalutationEnum.java
package com.philips.cdp.di.iap.model;

/**
 * Created by philips on 2/7/19.
 */

public enum SalutationEnum {

    MR("Mr."),
    MS("Ms.");

    private String field ;


    SalutationEnum(String field){
        this.field = field;
    }

    public String getField(){
        return this.field;
    }

}

File: Source/mec/Source/Library/mec/src/main/java/com/philips/cdp/di/mec/screens/address/SalutationEnum.kt
package com.philips.cdp.di.mec.screens.address

import com.philips.cdp.di.mec.R

enum class SalutationEnum(private val requestString:String, private val stringID:Int) {

    MR("Mr.", R.string.mec_mr),
    MS("Ms.",R.string.mec_mrs);

    fun getStringID(lRequestString:String):Int{

        for (value in values()){

           if(lRequestString.equals(value.requestString,true)){
               return value.stringID
           }
        }
        return MR.stringID
    }

    fun getRequestString(lStringID:Int):Int{

        for (value in values()){
            if(lStringID == value.stringID){
                return value.stringID
            }
        }
        return MR.stringID
    }
}